src/VisualStudio/Core/Impl/CodeModel/ProjectCodeModel.cs

```diff
@@ -110,5 +109,12 @@ public void OnSourceFileRenaming(string filePath, string newFilePath)
 
         EnvDTE.FileCodeModel IProjectCodeModel.GetOrCreateFileCodeModel(string filePath, object parent)
             => this.GetOrCreateFileCodeModel(filePath, parent).Handle;
+
+        public EnvDTE.FileCodeModel CreateFileCodeModel(SourceGeneratedDocument sourceGeneratedDocument)
+        {
+            // Unlike for "regular" documents, we make no effort to cache these between callers or hold them for longer lifetimes with
+            // events.
+            return FileCodeModel.Create(GetCodeModelCache().State, parent: null, sourceGeneratedDocument.Id, isSourceGeneratorOutput: true, new TextManagerAdapter()).Handle;
```

Comment:
do you want to CWT this?

Reply:
I could, I didn't want to do any caching at all just to keep things simple.